webxr/resources/webxr_util.js

```diff
@@ -39,7 +39,22 @@ function xr_session_promise_test(
 
   xr_promise_test(
       name,
-      (t) =>{
+      (t) => {
+          // Ensure that any pending sessions are ended and devices are
+          // disconnected when done. This needs to use a cleanup function to
+          // ensure proper sequencing. If this were done in a .then() for the
+          // success case, a test that expected failure would already be marked
+          // done at the time that runs, and the shutdown would interfere with
+          // the next test which may have started already.
+          t.add_cleanup(() => {
+                // If a session was created, end it.
+                if (testSession) {
+                  testSession.end().catch(() => {});
+                }
+                // Cleanup system state.
+                navigator.xr.test.disconnectAllDevices();
```

Comment:
This function should return a Promise that tracks the two asynchronous operations initiated here so that the harness waits for them to complete.

Reply:
Indeed, changed to use async/await. (The current chromium wrapper in webxr-test.js handles these actions synchronously, so I didn't notice the inconsistency.)

Reply:
Ah, got it. Just to be sure: using `await` like that will schedule the operations to occur in series, while they occur in parallel in `master` (and in the prior version of this patch). Is that a problem? If `testSession.end()` fails, should `disconnectAllDevices` run anyway?